lib/api/websiteGet.js

```diff
@@ -0,0 +1,231 @@
+import { errors } from 'arsenal';
+
+import constants from '../../constants';
+import metadata from '../metadata/wrapper';
+import bucketShield from './apiUtils/bucket/bucketShield';
+import { isObjAuthorized } from './apiUtils/authorization/aclChecks';
+import collectResponseHeaders from '../utilities/collectResponseHeaders';
+import validateHeaders from '../utilities/validateHeaders';
+
+/**
+ * _findRoutingRule - find applicable routing rule from bucket metadata
+ * @param {RoutingRule []} routingRules - array of routingRule objects
+ * @param {string} key - object key
+ * @param {number} [errCode] - error code to match if applicable
+ * @return {object | undefined} redirectInfo -- comprised of all of the
+ * keys/values from routingRule.getRedirect() plus
+ * a key of prefixFromRule and a value of routingRule.condition.keyPrefixEquals
+ */
+function _findRoutingRule(routingRules, key, errCode) {
+    if (!routingRules || routingRules.length === 0) {
+        return undefined;
+    }
+    // For AWS compat:
+    // 1) use first routing rules whose conditions are satisfied
+    // 2) for matching prefix no need to check closest match.  first
+    // match wins
+    // 3) there can be a match for a key condition with and without
+    // error code condition but first one that matches will be the rule
+    // used. So, if prefix foo without error and first rule has error condition,
+    // will fall through to next foo rule.  But if first foo rule has
+    // no error condition, will have match on first rule even if later
+    // there is more specific rule with error condition.
+    for (let i = 0; i < routingRules.length; i++) {
+        const prefixFromRule =
+            routingRules[i].getCondition().keyPrefixEquals;
+        const errorCodeFromRule =
+            routingRules[i].getCondition().httpErrorCodeReturnedEquals;
+        if (prefixFromRule !== undefined) {
+            if (!key.startsWith(prefixFromRule)) {
+                // no key match, move on
+                continue;
```

Comment:
Could these two if conditions be combined since there's no return? Would save a line :O